server/src/main/java/org/elasticsearch/index/mapper/FieldNamesFieldMapper.java

```diff
@@ -208,16 +204,15 @@ public void postParse(ParseContext context) {
     }
 
     @Override
-    public void parse(ParseContext context) throws IOException {
+    protected void parseCreateField(ParseContext context) throws IOException {
         // Adding values to the _field_names field is handled by the mappers for each field type
     }
 
     static Iterable<String> extractFieldNames(final String fullPath) {
```

Comment:
Given that this method is now only called from FieldMapper#createFieldNamesField(), maybe it should be moved to FieldMapper directly?

Reply:
I liked having it here because it feels like a specialty of `FieldNamesFieldMapper`, and it's natural to test it in `FieldNamesFieldMapperTests`.